upt_macports/upt_macports.py

```diff
@@ -51,6 +51,26 @@ def run_depends(self):
     def test_depends(self):
         return self._depends('test')
 
+    @property
+    def archive_type(self):
+        archive_url = self.upt_pkg.get_archive().url
+        if archive_url.endswith('.tar.gz'):
+            return 'targz'
+        elif archive_url.endswith('.7z'):
+            return '7z'
+        elif archive_url.endswith('.tar.bz2'):
+            return 'bzip2'
+        elif archive_url.endswith('.lzma'):
+            return 'lzma'
+        elif archive_url.endswith('.tar'):
+            return 'tar'
+        elif archive_url.endswith('.zip'):
+            return 'zip'
+        elif archive_url.endswith('.tar.xz'):
+            return 'xz'
+        else:
+            return 'undefined'
+
```

Comment:
This is fine, but maybe we could do something like;:

```
mappings = {
    '.tar.gz': 'targz',
    '.7z': '7z',
    ...
}
...
for extension, fmt in mappings.items():
    if archive_url.endswith(extension):
        return fmt
return 'undefined'
```

What do you think? There might be better variable names :)

Reply:
If using just the last `.` is fine then maybe we can do just this :

```
def archive_type(self):
        archive_keyword = {
            'gz': 'targz',
            '7z': '7z',
            'bz2': 'bzip2',
            'lzma': 'lzma',
            'tar': 'tar',
            'zip': 'zip',
            'xz': 'xz'
        }
        archive_url = self.upt_pkg.get_archive().url
        archive_type = archive_url.split('.')[-1]

        return archive_keyword.get(archive_type, 'unknown')
```

Reply:
Yeah, obviously using dict.get() is better :)

Reply:
Agreed! For the sake of consistency I would just use ```'gz': 'gz'``` as this is no different than for example the situation for ```xz```.

Reply:
I'm confused, don't we have a special case for "tar.gz" files?

Reply:
Yes, we do.

What I mean is that what is returned shouldn't be ```targz``` in my opinion (if we follow that logic ```xz``` should return ```tarxz```. So what I am proposing is -for consistency - to change ```'gz': 'targz',``` to ```'gz': 'gz',``` and in the template use ```{% elif pkg.archive_type != 'gz' %}```